libnopegl/src/scene.c

```diff
@@ -49,11 +54,13 @@ static void detach_root(struct ngl_scene *s)
 
 struct ngl_scene *ngli_scene_dup(struct ngl_scene *s)
 {
-    struct ngl_scene *copy = ngli_memdup(s, sizeof(*s));
+    struct ngli_scene *copy = ngli_memdup(s->internal, sizeof(*copy));
```

Comment:
can you use some `src`/`dst` prefixes over there ? something like dst_{public,private} and starts by storing `s->internal` in local variable so you can use its sizeof.

Reply:
this code goes away later on in the PR; the dup function is replaced by a ref function that looks like this:
```c
struct ngl_scene *ngl_scene_ref(struct ngl_scene *s)
{
    struct ngli_scene *priv = s->internal;
    NGLI_RC_REF(priv);
    return s;
}
```

Reply:
The commit has been dropped with your new iteration, so resolved :)